tests/agent/python/flaskapp/smoke.py

```diff
@@ -0,0 +1,258 @@
+        assert context['request']['url']['pathname'] in ("/foo", "/bar"), context['request']['url']['pathname']
+        assert context['request']['url']['hostname'] == 'localhost'
+
+        if lang == 'node':
+            assert context['response']['status_code'] == 200, context['response']['status_code']
+            assert context['user'] == {}, context
+            assert context['custom'] == {}, context
+
+        assert lookup(context, 'app', EXPECTATIONS[lang]['lang_key'], 'name') == lang, context
+        assert lookup(context, 'app', 'name') == 'test-app', context
+
+        assert lookup(context, 'app', 'agent', 'name') == EXPECTATIONS[lang]['agent_name'], context
+        assert lookup(context, 'app', 'framework', 'name') == EXPECTATIONS[lang]['framework'], context
+
+        assert context['tags'] == {}, context
+
+        assert hit['_source']['processor'] == {'name': 'transaction', 'event': 'transaction'}
+
+    traces_query = es_query("processor.event", "trace")
+    for hit in lookup(es.search(index, body=traces_query), 'hits', 'hits'):
+        context = lookup(hit, '_source', 'context')
+        assert lookup(context, 'app', 'name') == 'test-app', context
+
+        trace = lookup(hit, '_source', 'trace')
+
+        start = lookup(trace, 'start', 'us')
+        assert not anomaly(start), start
+
+        duration = lookup(trace, 'duration', 'us')
+        assert not anomaly(duration), duration
+
+        transaction_name, transaction_duration = transaction_dict[trace['transaction_id']]
+        assert duration < transaction_duration * 10, \
+            "trace duration {} is more than 10X bigger than transaction duration{}".format(duration, transaction_duration)
+
+        stacktrace = trace['stacktrace']
+        assert 15 < len(stacktrace) < 30, \
+            "number of frames not expected, got {}, but this assertion might be too strict".format(len(stacktrace))
+
+        fns = [frame['function'] for frame in stacktrace]
+        assert all(fns), fns
+        for attr in ['abs_path', 'line', 'filename']:
+            assert all(frame.get(attr) for frame in stacktrace), stacktrace[0].keys()
+
+        if trace['name'] == 'app.bar':
+            assert transaction_name == 'GET /bar', transaction_name
+            if lang == 'python':
+                assert trace['id'] == 0, trace['id']
+            assert 'bar_route' in fns
+        elif trace['name'] == 'app.foo':
+            assert transaction_name == 'GET /foo', transaction_name
+            if lang == 'python':
+                assert trace['id'] == 0, trace['id']
+            assert 'foo_route' in fns
+        else:
+            assert False, "trace name not expected {}".format(trace['name'])
+
+
+def lookup(d, *keys):
+    d1 = copy.deepcopy(d)
+    for k in keys:
+        d1 = d1[k]
+    return d1
+
+
+def es_query(field, val):
+    return {"query": {"term": {field: val}}}
+
+
+def reset():
+    f = '../../../../_meta/kibana/default/index-pattern/apmserver.json'
```

Comment:
This is the kibana index pattern you are using, I think you should consider using `apm-server/apm-server.template-es.json` which is created when the `make update` is run.

Reply:
hmmm not for me, seems like. i cant find that file after running `make update`

Reply:
you're right, it is not part of the `make update`, you have to run `make index-template` to get it.